README.md

```diff
@@ -174,3 +174,93 @@ Then create a new release:
+├── MyToken.sol
+└── zeppelin
+    ├── lifecycle
+    │   └── Destructible.sol
+    └── ownership
+        └── Ownable.sol
+
+```
+
+
+- Each even has its own contract. `AbstractConference.sol` is the superclass which contains all the functions. 
+- The `AbstractConference.sol` inherits from Distructible, Ownnable and GroupAdmin.
+- `ERC20Conference.sol` and `EthConference.sol` inherits `AbstractConference.sol` and override logic to RSVP and withdraw.
+- `Deployer.sol` is a factory contract to deploy smart contract for each event.
+- Due to the gas size exceeding block size, the actual binaries of the contracts are deployed via `ERC20Deployer.sol` and `EthDeployer.sol` and their deployed contract addresses are passed into the constructor of `Deployer.sol`. The `Deployer.sol` determins whether it should deploy ERC20 version if contract address is passed. Otherwise it will deploy the Eth version.
+- For ERC20 token contracts, we import openzeppelin library.
+
+```
+$grep openzeppelin contracts/*sol
+contracts/ERC20Conference.sol:import 'openzeppelin-solidity/contracts/token/ERC20/IERC20.sol';
+contracts/MyToken.sol:import 'openzeppelin-solidity/contracts/token/ERC20/ERC20.sol';
+contracts/MyToken.sol:import 'openzeppelin-solidity/contracts/token/ERC20/ERC20Detailed.sol';
+```
+
+The reason we have `zeppelin/*` contracts locally rather than importing from NPM is that they are as follows.
+
+- `zeppelin/lifecycle/Destructible.sol` no longer exists on openzeppelin repo
+- `zeppelin/lifecycle/Ownable.sol` has code changed (`owner` field is changed from `private` to `public`)
+
+### Some decision behind the architectural choice.
+
+#### Deploying each contract per event.
+
+A new contract needs to be deployed for each event, which incurs some cost at each deploy.
+This makes it relatively expensive to use for a small number of participant especially when the gas price or Ether price is expensive. However, this will allow us to constantly update and refactor the contract to be up to date without having complex upgradability strategy.
+
+#### No strict time dependencies.
+
+The current contract does not contain any information about the event start and ends time as real events often do not have strict deadlines on participation. Also, the current unstable nature of Ethereum mainnet occasionally make it difficult (or too expensive) to interact with the contract in a timely manner so it often comes down to the event owner to decide when it ends the event. The only exception is `cooling period` which last for a week by default. Having the cooling period is mainly to encourage users to withdraw deposits as soon as possible so that users can avoid possible loss of funds if any bugs or vulnerabilities are found (or the deployment account is compromised). Please refer to [this blog post](https://medium.com/@makoto_inoue/running-everyday-dapp-when-ethereum-is-under-pressure-2c5bf4412c13) for the user impact when Ethereum network is under performance pressure.
```

Comment:
start and end* time
occasionally makes* it difficult
event owner to decide when one* (or use he/she) ends the event.
exception is `cooling period` which lasts* for
user impact when the* Ethereum network